docs/EN_US/HPCCClientTools/CT_Mods/CT_ECL_CLI.xml

```diff
@@ -6298,6 +6298,184 @@ ecl zapgen W20171018-091399 --path ~ --inc-thor-slave-logs --description "Unexpe
+        </informaltable>
+
+        <para>The <emphasis>ecl roxie memlock</emphasis> command locks Roxie
+        heap memory. This helps to prevent intermittent slow queries by
+        ensuring that the operating system does not swap the memory
+        out.</para>
+
+        <para>Examples:</para>
+
+        <programlisting>ecl roxie memlock mycluster
+</programlisting>
+      </sect2>
+
+      <sect2 id="CT-RCTL-MEMUNLOCK" role="brk">
+        <title>ecl roxie memunlock</title>
+
+        <para><emphasis role="bold">ecl roxie memunlock
+        &lt;process_cluster&gt;</emphasis></para>
+
+        <informaltable colsep="1" frame="all" rowsep="1">
+          <tgroup cols="2">
+            <colspec colwidth="80.50pt" />
+
+            <colspec />
+
+            <tbody>
+              <row>
+                <entry><emphasis>Options</emphasis></entry>
+
+                <entry></entry>
+              </row>
+
+              <row>
+                <entry><emphasis>process_cluster</emphasis></entry>
+
+                <entry>The roxie process cluster to memunlock</entry>
+              </row>
+            </tbody>
+          </tgroup>
+        </informaltable>
+
+        <para>The <emphasis>ecl roxie memunlock</emphasis> command unlocks
+        Roxie heap memory.</para>
+
+        <para>Examples:</para>
+
+        <programlisting>ecl roxie memunlock mycluster
+</programlisting>
+      </sect2>
+
+      <sect2 id="CT-RCTL-GETMEMLOCK" role="brk">
+        <title>ecl roxie getmemlocked</title>
+
+        <para><emphasis role="bold">ecl roxie getmemlocked
+        &lt;process_cluster&gt;</emphasis></para>
+
+        <informaltable colsep="1" frame="all" rowsep="1">
+          <tgroup cols="2">
+            <colspec colwidth="80.50pt" />
+
+            <colspec />
+
+            <tbody>
+              <row>
+                <entry><emphasis>Options</emphasis></entry>
+
+                <entry></entry>
+              </row>
+
+              <row>
+                <entry><emphasis>process_cluster</emphasis></entry>
+
+                <entry>The Roxie process cluster to getmemlocked
+                status</entry>
+              </row>
+            </tbody>
+          </tgroup>
+        </informaltable>
+
+        <para>The <emphasis>ecl roxie getmemlocked</emphasis> command
+        retrieves the Roxie heap memory lock status. You can run the ECL
+        command-line on any host as long as it knows what ESP to connect to.
+        This is the ‘best’ way to check if Roxie memory is locked.</para>
+
+        <para>Examples:</para>
+
+        <programlisting>ecl roxie getmemlocked mycluster
+</programlisting>
+
+        <sect3 id="CT-CheckMemIndivNodes">
+          <title>Checking individual nodes</title>
+
+          <para>The above command <emphasis>ecl roxie getmemlocked</emphasis>
+          is the best way to check if Roxie memory is locked, however, if you
+          wanted to check it out on each individual Roxie node, you can use
```

Comment:
and you can log onto the roxie node,...

I'm not sure if this should be in the documentation or not.  Possibly in the admin hints, but not in the client tools.  On a cloud (or well locked down bare-metal) system you are unlikely to be able to log onto the nodes.